modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java

```diff
@@ -1280,17 +1280,16 @@ private int partition(CacheObjectContext ctx, @Nullable GridCacheContext cctx, O
         @Nullable IncompleteCacheObject incompleteObj,
         boolean createCacheObjectShadow
     ) {
-        if (incompleteObj == null) {
+        if (incompleteObj == null)
             incompleteObj = createCacheObjectShadow ? new IncompleteCacheObjectShadow(buf) : new IncompleteCacheObject(buf);
```

Comment:
If the exact buffer size is not important, you can set the `IncompleteCacheObjectShadow` to `ready` without reading all data pages. This will make the estimation of big values much faster and decrease the probability of page replacement.